ariac_sensors/src/ariac_logical_camera_plugin.cpp

```diff
@@ -120,6 +123,8 @@ void AriacLogicalCameraPluginPrivate::OnUpdate()
 
   const auto & image = this->sensor_->Image();
 
+  gazebo::common::Time sensor_update_time = this->sensor_->LastMeasurementTime();
```

Comment:
I would suggest you print sensor_update_time to see if the correct value is assigned here. This seems an issue which can be resolved with a static cast.

Reply:
The `LastMeasurementTime()` function apparently doesn't work for logical cameras. I have switched to ros time from the node and it is working now.